itou/siae_evaluations/models.py

```diff
@@ -399,27 +399,30 @@ def review(self):
         ACCEPTED = evaluation_enums.EvaluatedSiaeState.ACCEPTED
         REFUSED = evaluation_enums.EvaluatedSiaeState.REFUSED
         ADVERSARIAL_STAGE = evaluation_enums.EvaluatedSiaeState.ADVERSARIAL_STAGE
+        NOTIFICATION_PENDING = evaluation_enums.EvaluatedSiaeState.NOTIFICATION_PENDING
         if self.state in {ACCEPTED, REFUSED, ADVERSARIAL_STAGE}:
-            adversarial_stage = bool(self.reviewed_at)
+            from_adversarial_stage = bool(self.reviewed_at)
+            previous_state = self.state
 
             now = timezone.now()
-            if self.state == ACCEPTED:
+            if previous_state == ACCEPTED:
                 self.reviewed_at = now
                 self.final_reviewed_at = now
-            elif self.state == REFUSED:
+            elif previous_state == REFUSED:
                 self.reviewed_at = now
-            elif self.state == ADVERSARIAL_STAGE:
+            elif previous_state == ADVERSARIAL_STAGE:
                 self.final_reviewed_at = now
             else:
                 raise TypeError(f"Cannot review an “{self.__class__.__name__}” with status “{self.state}”.")
             self.save()
-
+            # Invalidate the cache, a review changes the state of the evaluation.
+            del self.state
```

Comment:
Peut être le mettre dans le `save()` directement ?

Reply:
J’ai un peu peur des effets de bords ce cette suggestion. Je l’ajoute à ma belle todo liste de “à changer dans le contrôle a posteriori”.

Reply:
Notamment, de la doc de cached_property :

> Because of the way the [descriptor protocol](https://docs.python.org/3/reference/datamodel.html#descriptor-invocation) works, using del (or delattr) on a cached_property that hasn’t been accessed raises AttributeError.